src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -1657,15 +1659,22 @@ public void CopyTo(Span<byte> destination)
         }
 
         /// <summary>
-        /// Sets the bytes in 'outBytes' to be random values.   outBytes.Length must be less than or equal to 16
+        /// Sets the bytes in 'outBytes' to be random values. outBytes.Length must be either 8 or 16 bytes.
         /// </summary>
         /// <param name="outBytes"></param>
         internal static unsafe void SetToRandomBytes(Span<byte> outBytes)
         {
-            Debug.Assert(outBytes.Length <= sizeof(Guid));     // Guid is 16 bytes, and so is TraceId
-            Guid guid = Guid.NewGuid();
-            ReadOnlySpan<byte> guidBytes = new ReadOnlySpan<byte>(&guid, sizeof(Guid));
-            guidBytes.Slice(0, outBytes.Length).CopyTo(outBytes);
+            Debug.Assert(outBytes.Length == 16 || outBytes.Length == 8);
```

Comment:
It could probably be made simpler by calling the interop method directly. Something like:

```c#
internal static unsafe void SetToRandomBytes(Span<byte> outBytes)
{
    Debug.Assert(outBytes.Length == 16 || outBytes.Length == 8);
    Interop.GetRandomBytes((byte*)outBytes.GetPinnableReference(), outBytes.Length);
}
```

Reply:
@am11 Using the interop will require to include the source file `Interop.Random.cs` to the project. That will require to build the assembly differently for OSX, Linux and Windows which we don't currently do. That will increase the package size too as we'll include more cross generated assemblies.

Reply:
Ah ok. I didn't looked at the proj file of this assmebly.